node/src/Pos/Block/Slog/Context.hs

```diff
@@ -1,40 +1,69 @@
--- | Functions operation on 'SlogContext'.
+-- | Functions operation on 'SlogContext' and its subtypes.
 
 module Pos.Block.Slog.Context
-       ( SlogContext (..)
+       ( SlogGState (..)
+       , SlogContext (..)
+
+       , mkSlogGState
        , mkSlogContext
-       , cloneSlogContext
+       , cloneSlogGState
        , slogGetLastSlots
        , slogPutLastSlots
        ) where
 
 import           Universum
 
-import           Pos.Block.Slog.Types  (HasSlogContext (..), LastBlkSlots,
-                                        SlogContext (..), scLastBlkSlots)
+import           Formatting            (int, sformat, (%))
+import qualified System.Metrics        as Ekg
+
+import           Pos.Block.Slog.Types  (HasSlogGState (..), LastBlkSlots,
+                                        SlogContext (..), SlogGState (..),
+                                        sgsLastBlkSlots)
+import           Pos.Core              (HasCoreConstants, blkSecurityParam,
+                                        fixedTimeCQSec)
 import           Pos.DB.Class          (MonadDBRead)
 import           Pos.GState.BlockExtra (getLastSlots)
+import           Pos.Reporting         (mkMetricMonitorState)
+
+-- | Make new 'SlogGState' using data from DB.
+mkSlogGState :: (MonadIO m, MonadDBRead m) => m SlogGState
+mkSlogGState = do
+    _sgsLastBlkSlots <- getLastSlots >>= newIORef
+    return SlogGState {..}
 
 -- | Make new 'SlogContext' using data from DB.
-mkSlogContext :: (MonadIO m, MonadDBRead m) => m SlogContext
-mkSlogContext = do
-    _scLastBlkSlots <- getLastSlots >>= newIORef
+mkSlogContext ::
+       (MonadIO m, MonadDBRead m, HasCoreConstants)
+    => Maybe Ekg.Store
```

Comment:
My advice actually would not be to make the store optional, e.g. for testing use cases or whatever. You can certainly make reporting the metrics optional, but it's simpler to always have a store and optionally report, than to also have to deal with optionally having a store.